jbmc/unit/java-testing-utils/require_goto_statements.cpp

```diff
@@ -295,6 +294,49 @@ const code_declt &require_goto_statements::require_declaration_of_name(
   throw no_decl_found_exceptiont(variable_name.c_str());
 }
 
+/// Follow the chain of non-null assignments until we find a symbol that
```

Comment:
This would be easier to explain if split into three functions: one that finds a unique non-null *expression* feeding a symbol, one that finds a unique *symbol*, and one which does the iteration. Then the first can be explained tersely that the input program may contain many "input = null" assignments and must contain exactly one "input = e"; returns e. The others are then easier to define in terms of that operation.